### PR TITLE
Feature/detekt

### PR DESCRIPTION
**Pull request by georgebaronheid**
Added detekt functionalities, its dependencies.gradle variables, config directory and within it detekt.gradle file and detekt-config.yml file.

### CODE REVIEW

File: dependencies.gradle [hunk 1]
```diff
@@ -14,17 +14,21 @@ ext {
             'grade_version'        : '4.0.1',
             'core_ktx_version'     : '1.3.1',
             'appcompat_version'    : '1.2.0',
+            'detekt_version'       : '1.11.0-RC2',
             'junit_version'        : '4.12',
             'test_junit_version'   : '1.1.1',
-            'espresso_core_version': '3.2.0'
+            'espresso_core_version': '3.2.0',
+            'ktlint_version'       : '0.36.0'
```
Comment (sp4wna1) [inline dependencies.gradle]: Remover o ktlint

File: dependencies.gradle [hunk 2]
```diff
@@ -14,17 +14,21 @@ ext {
...
 
     libraries = [
             'gradle'              : "com.android.tools.build:gradle:${versions.grade_version}",
             'kotlin_gradle_plugin': "org.jetbrains.kotlin:kotlin-gradle-plugin:${versions.kotlin_version}",
             'kotlin_stdlib'       : "org.jetbrains.kotlin:kotlin-stdlib:${versions.kotlin_version}",
             'core_ktx'            : "androidx.core:core-ktx:${versions.core_ktx_version}",
-            'appcompat'           : "androidx.appcompat:appcompat:${versions.appcompat_version}"
+            'appcompat'           : "androidx.appcompat:appcompat:${versions.appcompat_version}",
+            'detekt'              : "io.gitlab.arturbosch.detekt:detekt-cli:${versions.detekt_version}",
+            'ktlint'              : "com.pinterest:ktlint:${versions.ktlint_version}"
```
Comment (sp4wna1) [inline dependencies.gradle]: Remover o ktlint

**Review by sp4wna1 — CHANGES_REQUESTED** [review]

File: dependencies.gradle (hunk 1 above)
Comment (georgebaronheid) [reply dependencies.gradle]: Done

File: dependencies.gradle (hunk 2 above)
Comment (georgebaronheid) [reply dependencies.gradle]: Done

**Review by sp4wna1 — APPROVED** [review]

**Pull request merged**